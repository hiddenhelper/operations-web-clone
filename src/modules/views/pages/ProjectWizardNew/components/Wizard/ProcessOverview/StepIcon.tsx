import React, { memo } from 'react';

import { getConditionalDefaultValue } from 'utils';
import { useStepIconStyles } from './StepIconStyles';

export interface IStepIconProps {
  active: boolean;
  completed: boolean;
  started: boolean;
}

const StepIcon = ({ active, completed, started }: IStepIconProps) => {
  const classes = useStepIconStyles();

  return (
    <div className={classes.root} data-testid={'step-icon'}>
      <div
        className={`${classes.circle} ${getConditionalDefaultValue(active && !completed, classes.active, '')} ${getConditionalDefaultValue(
          completed,
          classes.completed,
          ''
        )} ${getConditionalDefaultValue(started && !completed, classes.started, '')} ${getConditionalDefaultValue(
          !active && !completed,
          classes.uncompleted,
          ''
        )} ${getConditionalDefaultValue(started && active && !completed, classes.startedAndActive, '')}`}
      />
      {completed ? <span className={classes.checkmark} /> : ''}
    </div>
  );
};

export default memo(StepIcon);
