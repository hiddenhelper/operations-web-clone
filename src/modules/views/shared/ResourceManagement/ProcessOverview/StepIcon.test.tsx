import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import StepIcon, { IStepIconProps } from './StepIcon';

describe('Step Icon', () => {
  let wrapper: RenderResult;
  let props: IStepIconProps;

  beforeEach(() => {
    props = {
      active: true,
      completed: false,
      started: false,
    };

    wrapper = render(<StepIcon {...props} />);
  });

  it('should render', () => {
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
