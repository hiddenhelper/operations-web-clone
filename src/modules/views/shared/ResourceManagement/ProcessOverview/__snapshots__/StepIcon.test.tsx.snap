// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step Icon should render 1`] = `
<body>
  <div>
    <div
      class="makeStyles-root-1"
      data-testid="step-icon"
    >
      <div
        class="makeStyles-circle-8 makeStyles-active-3    "
      />
      
    </div>
  </div>
</body>
`;
